client/src/components/ScoreForm.js

```diff
@@ -28,23 +42,56 @@ const TableExercises = ({ exercises }) => (
 			</Table.Row>
 		</Table.Header>
 		<Table.Body>
-			{exercises.map(exercise => {
-				const { name,value,units,type } = {...exercise}
-				return (
-					<Table.Row key={name}>
-						<Table.Cell>{name}</Table.Cell>	
-						{type === 'interval' && <Table.Cell><CurrIntegerExercise value={value} units={units}/></Table.Cell>}
-						{type === 'interval' && <Table.Cell><NewIntegerExercise value={value} units={units}/></Table.Cell>}
-						{type === 'timer' && <Table.Cell><CurrTimerExercise value={value}/></Table.Cell>}
-						{type === 'timer' && <Table.Cell><NewTimerExercise value={value}/></Table.Cell>}
-					</Table.Row>
-				)
-			})}	
+				{exercises.map((exercise, index) => {
+					const { name, units, type } = {...exercise}
+					let fieldName = []
+					if (type === 'timer') {
+						fieldName[0] = 'user.exercises[' + index + '].value.hr'
+						fieldName[1] = 'user.exercises[' + index + '].value.min'
+					} else if (type === 'interval') {
+						fieldName = 'user.exercises[' + index + '].value'
+					}
+					return (
+						<Table.Row key={index}>
+							<Table.Cell>{name}</Table.Cell>	
+							{type === 'interval' && <Table.Cell><CurrIntegerExercise currValue={currValues[index].value} units={units} /></Table.Cell>}
```

Comment:
A bit nitpicky, but it's a bit difficult to read the code. Some suggestions - you can move some code on new lines lines so it has the same indentation with the content the other Table.Cells; like so below

```
{type === 'interval' && 
<Table.Cell>
    <CurrIntegerExercise currValue={currValues[index].value} units={units} />
</Table.Cell>
}
```

Or you can define a function that will render the cell components and place that code further outside the actual ```render``` fn, and call fn that within ```render```

Reply:
Updated code with indentation. Let me know if this is better.